File: wasm_first/sort_random_array.cpp
#include <iostream>
#include <stdlib.h>
#include <vector>
#include <chrono>
#include <algorithm>

using namespace std;

int main(){

    srand(time(NULL));

    auto start = chrono::high_resolution_clock::now();

    const int len = 10000000;
    vector<int> arr(len);
    for(int i = 0; i < len; i++)
    {
        arr[i] = rand() % len;
    }
    
    auto end = chrono::high_resolution_clock::now();
    auto duration = chrono::duration_cast<chrono::milliseconds>(end - start).count();

    cout << "Array generated in " << duration << " ms" << endl;

    start = chrono::high_resolution_clock::now();
    
    sort(arr.begin(), arr.end());

    end = chrono::high_resolution_clock::now();
    duration = chrono::duration_cast<chrono::milliseconds>(end - start).count();

    cout << "Array sorted in " << duration << " ms" << endl;

    return 0;
}

File: wasm_first/sort_random_array.js
let arr = []
console.time('Create random array');
for (let index = 0; index < 10000000; index++) {
    arr[index] = Math.floor(Math.random * 1e6);
}
console.timeEnd('Create random array');
console.time('Sort random array');

arr.sort((a, b) => a - b);
console.timeEnd('Sort random array');